packages/terra-tabs/src/_CollapsibleTabs.jsx

```diff
@@ -0,0 +1,261 @@
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ResizeObserver from 'resize-observer-polyfill';
+import 'terra-base/lib/baseStyles';
+import TabUtils from './TabUtils';
+import Menu from './_TabMenu';
+import styles from './Tabs.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Key of the current active tab.
+   */
+  activeKey: PropTypes.string.isRequired,
+
+  /**
+   * Index of the current active tab.
+   */
+  activeIndex: PropTypes.number.isRequired,
+
+  /**
+   * Tabs to be displayed in the collapsible tab bar.
+   */
+  children: PropTypes.node.isRequired,
+
+  /**
+   * Tabs style. One of: "modular-centered", "modular-left-aligned", or "structural".
+   */
+  variant: PropTypes.oneOf(['modular-centered', 'modular-left-aligned', 'structural']).isRequired,
+
+  /**
+   * Callback function when selection has changed.
+   * Parameters: 1. Event 2. Selected pane element
+   */
+  onChange: PropTypes.func.isRequired,
+
+  /**
+   * Callback function when label truncation state has changed.
+   * Parameters: 1. Bool indicating if any of the tab labels have been truncated.
+   */
+  onTruncationChange: PropTypes.func,
+};
+
+class CollapsibleTabs extends React.Component {
+  constructor(props) {
+    super(props);
+    this.setContainer = this.setContainer.bind(this);
+    this.setMenuRef = this.setMenuRef.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+    this.handleSelectionAnimation = this.handleSelectionAnimation.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.handleFocusLeft = this.handleFocusLeft.bind(this);
+    this.handleFocusRight = this.handleFocusRight.bind(this);
+    this.state = {
+      hiddenStartIndex: -1,
+      menuHidden: false,
+      isCalculating: true,
+    };
+  }
+
+  componentDidMount() {
+    this.resizeObserver = new ResizeObserver((entries) => {
+      // Resetting the state so that all elements will be rendered face-up for width calculations
+      if (this.state.hiddenStartIndex !== -1 || this.state.menuHidden || !this.state.isCalculating) {
+        this.setState({ hiddenStartIndex: -1, menuHidden: false, isCalculating: true });
+      }
+      this.handleResize(entries[0].contentRect.width);
+    });
+    this.resizeObserver.observe(this.container);
+
+    this.handleSelectionAnimation();
+  }
+
+  componentDidUpdate() {
+    this.handleSelectionAnimation();
+  }
+
+  componentWillUnmount() {
+    this.resizeObserver.disconnect(this.container);
+    this.container = null;
+  }
+
+  setContainer(node) {
+    if (node === null) { return; } // Ref callbacks happen on mount and unmount, element will be null on unmount
+    this.container = node;
+  }
+
+  setMenuRef(node) {
+    if (node === null) { return; }
+    this.menuRef = node;
+  }
+
+  handleResize(width) {
+    const menuMarginLeft = parseInt(window.getComputedStyle(this.menuRef, null).getPropertyValue('margin-left'), 10);
```

Comment:
should these be rem values?

Reply:
The `10` here is the `radix` parameter for `parseInt()` and indicates that I want the value to be parsed based on the decimal number system.